paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/commands/WriteIntoPaimonTable.scala

```diff
@@ -74,13 +75,13 @@ case class WriteIntoPaimonTable(
     val primaryKeyCols = tableSchema.trimmedPrimaryKeys().asScala.map(col)
     val partitionCols = tableSchema.partitionKeys().asScala.map(col)
 
-    val dataEncoder = RowEncoder.apply(data.schema).resolveAndBind()
+    val dataEncoder = ExpressionEncoder(encoderFor(data.schema)).resolveAndBind()
     val originFromRow = dataEncoder.createDeserializer()
 
     // append _bucket_ column as placeholder
     val withBucketCol = data.withColumn(BUCKET_COL, lit(-1))
     val bucketColIdx = withBucketCol.schema.size - 1
-    val withBucketDataEncoder = RowEncoder.apply(withBucketCol.schema).resolveAndBind()
+    val withBucketDataEncoder = ExpressionEncoder(encoderFor(withBucketCol.schema)).resolveAndBind()
```

Comment:
Can you check if this work for Spark 3.3 too?

Reply:
@JingsongLi, above implementation doesn't support Spark 3.3. I have used reflect way to solve compatibility problem and checked Spark 3.3 successful. PTAL.